### PR TITLE
sys/chunked_ringbuffer: model Kconfig

### PR DESCRIPTION
**Pull request by leandrolanzieri**
### Contribution description
This models `chunked_ringbuffer` in Kconfig, and also moves a configuration option to the `CONFIG` namespace.

### Testing procedure
- Green CI
- Check the modelling

### Issues/PRs references
Split from #17739

### CODE REVIEW

File: sys/include/chunked_ringbuffer.h [hunk 1]
```diff
@@ -25,19 +25,32 @@
...
+#ifdef DOXYGEN
 #define CHUNK_NUM_MAX   (4)
 #endif
 
+/**
+ * @brief The maximum number of chunks that can be stored in a Chunked Ringbuffer
+ *
+ */
+#ifndef CONFIG_CHUNK_NUM_MAX
+#  ifdef CHUNK_NUM_MAX
```
Comment (benpicco) [inline sys/include/chunked_ringbuffer.h:47]: No need for the backcompat, this module was just recently added

Comment (leandrolanzieri) [reply sys/include/chunked_ringbuffer.h]: OK, removed it!

**Review by benpicco — APPROVED** [review]

**Comment by leandrolanzieri** [comment]
All green here!

**Comment by leandrolanzieri** [comment]
Thanks for the review!

**Pull request merged**